File: app/src/main/java/com/syh/framework/list/MyQuickAdapter.java
package com.syh.framework.list;

import android.widget.TextView;


import com.syh.framework.R;

import java.util.HashMap;
import java.util.List;

/**
 * Created bg shenyonghe on 2018/6/8.
 */
public class MyQuickAdapter extends BaseQuickAdapter<ItemDemo> {

    private static final HashMap<Integer, Integer> map;

    static {
        map = new HashMap<>();
        map.put(0, R.layout.item_test_list);
        map.put(1, R.layout.item_test2_list);
    }

    public MyQuickAdapter(List<ItemDemo> data) {
        super(map, data);
    }

    @Override
    public int getItemViewType(int position) {
        return 1;
    }

    @Override
    public void convert(BaseViewHolder holder, ItemDemo s) {
        TextView title = holder.getView(R.id.tv_title);
        title.setText(s.getName());
    }
}

File: app/src/main/java/com/syh/framework/list/MyQuickAdapter.kt
package com.syh.framework.list

import android.app.Activity
import android.content.Intent
import android.widget.TextView
import com.syh.framework.R
import com.syh.framework.expose.attachExposeKey
import com.syh.framework.test.SPActivity
import java.util.*

/**
 * Created bg shenyonghe on 2018/6/8.
 */
class MyQuickAdapter(data: List<ItemDemo?>?) : BaseQuickAdapter<ItemDemo?>(map, data) {
    companion object {
        private var map: HashMap<Int, Int>? = null

        init {
            map = HashMap()
            map!![0] = R.layout.item_test_list
            map!![1] = R.layout.item_test2_list
        }
    }

    override fun getItemViewType(position: Int): Int {
        return 1
    }

    override fun convert(holder: BaseViewHolder?, s: ItemDemo?) {
        var context =  holder!!.itemView.context as Activity
        holder!!.itemView.attachExposeKey(s!!.name)
        val title = holder!!.getView<TextView>(R.id.tv_title)
        title.text = s!!.name
        holder!!.itemView.setOnClickListener {
            context.startActivity(Intent(context, SPActivity::class.java))
        }
    }

}